### DISCUSSION TITLE
Can I get the source name?

### DISCUSSION THREAD

**Discussion by renatoianhez** [pschatzmann/ESP32-A2DP#461] [Q&A] [answered]
I would like to show on a screen the name of the source of the bluetooth transmission, such as "John's Cell Phone" or "Mary's iPhone". Is this possible?

**Comment by pschatzmann** [comment, marked as answer]
Yes it is. 
I suggest you go thru the available method names in the class documentation. Hint search for name

**Comment by renatoianhez** [comment]
Oh, thanks!
I have doubts in the implementation of this. The code snippet I use is below:
```
  void avrc_metadata_callback(uint8_t id, const uint8_t *text) {
  Serial.printf("==> AVRC metadata rsp: attribute id 0x%x, %s\n", id, text);
  tela.setFreeFont(FF5);
  tela.fillScreen(TFT_BLACK);
  tela.fillRect(0, 0, 160, 28, TFT_WHITE);
  tela.setCursor(28, 18);
  tela.setTextColor(TFT_BLUE); //texto azul
  tela.println("BLUETOOTH");
  tela.setTextFont(2);
  tela.setCursor(0, 30);
  switch (id) {
    case ESP_AVRC_MD_ATTR_TITLE:
      tituloBT = (char*) text;
      break;
    case ESP_AVRC_MD_ATTR_ARTIST:
      artistaBT = (char*) text;
      break;
    case ESP_AVRC_MD_ATTR_ALBUM:
      albumBT = (char*) text;
      break;
  }
  const char nomeBT = a2dp_sink.get_peer_name();
  tela.setTextColor(TFT_WHITE);
  tela.print("Conectado a: ");
  tela.println(nomeBT);
  tela.setTextColor(TFT_PINK);
  tela.println(tituloBT);
  tela.setTextColor(TFT_YELLOW);
  tela.println(artistaBT);
  tela.setTextColor(TFT_LIGHTGREY);
  tela.println(albumBT);
}
```
When I try to compile it gives this error: 'class BluetoothA2DPSink' has no member named 'get_peer_name'
Where did I go wrong?

**Reply by renatoianhez** [reply]
I achieved! I did the following:
`tela.println(a2dp_sink.get_connected_source_name());`
Thanks!

**Comment by pschatzmann** [comment]
get_connected_source_name() should work with your release. 
If you upgrade to the latest version get_peer_name() will work as well.
